eppo/src/androidTest/java/cloud/eppo/android/EppoClientTest.java

```diff
@@ -230,9 +229,12 @@ private int runTestCaseFileStream(InputStream testCaseStream) throws IOException
                 assertEquals(expectedBooleanAssignments, actualBooleanAssignments);
                 return actualBooleanAssignments.size();
             case JSON:
-                List<String> actualJSONAssignments = this.getJSONAssignments(testCase);
-                assertEquals(testCase.expectedAssignments, actualJSONAssignments);
-                return actualJSONAssignments.size();
+                // test parsed json
+                List<JsonElement> actualParsedJSONAssignments = this.getJSONAssignments(testCase);
+                List<String> actualJSONStringAssignments = actualParsedJSONAssignments.stream().map(x -> x.toString()).collect(Collectors.toList());
+
+                assertEquals(testCase.expectedAssignments, actualJSONStringAssignments);
+                return actualParsedJSONAssignments.size();
```

Comment:
Since we know that `getJSONStringAssignment()` calls `getParsedJSONAssignment()` under the hood, you could just call that and check equality, and we'd be implicitly testing the other as well. A bit of a unit testing gray area as it relies on implementation knowledge and wouldn't protect us if the implementation changes. But right now `getJSONStringAssignment()` is totally untested, so I think it could be a better alternative.

Another option could be to just also call and collect the results of `getJSONStringAssignment()` and just assert both in this `case JSON:` block. That may be the best approach.

Reply:
Thanks ya I had gone that direction originally- trouble I ran into is that the "expectedAssignments" is stringified. I'll give it another shot to see if we can figure out a good pattern to test both json functions.